src/repl/scala/tools/nsc/interpreter/session/FileBackedHistory.scala

```diff
@@ -80,5 +80,5 @@ object FileBackedHistory {
   import Properties.userHome
 
   def defaultFileName = ".scala_history"
-  def defaultFile: File = File(Path(userHome) / defaultFileName)
+  def defaultFile: File = File( Path( Properties.propOrElse( "scala.repl.histfile", userHome +"/"+ defaultFileName) ) )
```

Comment:
Using `+ "/" +` to create the path fails to account for Windows users and their backslashes. That's what the `/` method on `Path` does. Maybe something like:
```scala
File(Properties.propOrNone("scala.repl.histfile").fold(Path(userHome) / defaultFileName))(Path(_)))
```